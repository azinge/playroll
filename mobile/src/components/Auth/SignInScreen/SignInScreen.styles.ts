import { StyleSheet } from "react-native";

export default StyleSheet.create({
  screenContainer: {
    flex: 1,
    backgroundColor: "#fff",
  },
  optionsContainer: {
    flex: 2,
    backgroundColor: "white",
  },
  usernameField: {
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
  },
  passwordContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  passwordField: {
    flex: 4,
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
  },
});
