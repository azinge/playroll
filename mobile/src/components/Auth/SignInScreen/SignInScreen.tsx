/**
 * SignInScreen
 */

import * as React from 'react';
import {
  ActivityIndicator,
  Text,
  View,
  Button,
  TouchableWithoutFeedback,
  TouchableOpacity,
  TextInput,
  Switch,
  Keyboard,
  SafeAreaView,
} from 'react-native';
import {
  NavigationScreenProp,
  StackActions,
  NavigationActions,
} from 'react-navigation';

import { SignInMutation } from '../../../graphql/requests/Auth';

import styles from './SignInScreen.styles';
import NavigationService from '../../../services/NavigationService';
import { Icon } from 'react-native-elements';

export interface Props {
  onLoginPress?: () => void;
  onLogoutPress?: () => void;
  navigation?: NavigationScreenProp<{}>;
}

interface State {
  username: string;
  password: string;
  showPassword: boolean;
  signedUp: boolean;
}

export default class SignInScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      showPassword: true,
      signedUp: true,
    };
    this.revealPassword = this.revealPassword.bind(this);
    this.toggleSignUp = this.toggleSignUp.bind(this);
  }

  revealPassword() {
    this.setState({ showPassword: !this.state.showPassword });
  }

  toggleSignUp() {
    NavigationService.navigate('SignUp');
  }

  renderSegueToSignIn() {
    return (
      <TouchableOpacity
        style={styles.segueToSignInContainer}
        onPress={() => {
          NavigationService.goBack();
        }}
      >
        <Icon
          name='arrow-back'
          type='material'
          color='#6A0070'
          onPress={() => {
            NavigationService.goBack();
          }}
        />
        <Text style={styles.signInTitle}>Sign In</Text>
      </TouchableOpacity>
    );
  }

  renderHeader() {
    return (
      <View style={styles.loginHeader}>
        <Text style={styles.loginText}>Log In</Text>
      </View>
    );
  }

  toggleSignIn(signIn) {
    signIn().then(
      () =>
        this.props.navigation &&
        this.props.navigation.dispatch(
          StackActions.reset({
            key: null,
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Main' })],
          })
        )
    );
  }

  renderSigninButton() {
    return (
      <SignInMutation
        variables={{
          username: this.state.username,
          password: this.state.password,
        }}
      >
        {/* What does error mean down here  */}
        {(signIn, { loading, error, data }) => {
          return (
            <TouchableOpacity
              onPress={() => this.toggleSignIn(signIn)}
              style={styles.submitButton}
            >
              {loading ? (
                <ActivityIndicator color={'white'} />
              ) : (
                <Text style={styles.submitButtonText}>Sign In</Text>
              )}
            </TouchableOpacity>
          );
        }}
      </SignInMutation>
    );
  }

  render() {
    return (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        <SafeAreaView style={styles.mainContainer}>
          <View style={styles.container}>
            {this.renderSegueToSignIn()}
            {this.renderHeader()}
            <TextInput
              placeholder='Username'
              autoCapitalize='none'
              style={styles.inputContainer}
              onChangeText={(username: string) => this.setState({ username })}
              value={this.state.username}
            />
            <TextInput
              placeholder='Password'
              secureTextEntry={this.state.showPassword}
              autoCapitalize='none'
              style={styles.inputContainer}
              onChangeText={(password: string) => this.setState({ password })}
              value={this.state.password}
            />
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              <Text style={styles.showPasswordText}>Show Password</Text>
              <Switch
                onValueChange={this.revealPassword}
                value={!this.state.showPassword}
                style={{ marginTop: 25, marginLeft: 10 }}
              />
            </View>
          </View>
          {this.renderSigninButton()}
        </SafeAreaView>
      </TouchableWithoutFeedback>
    );
  }
}
