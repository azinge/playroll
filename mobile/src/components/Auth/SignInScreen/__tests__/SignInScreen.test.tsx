import * as React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';

import SignInScreen from '../SignInScreen';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MockedProvider>
        <SignInScreen />
      </MockedProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
