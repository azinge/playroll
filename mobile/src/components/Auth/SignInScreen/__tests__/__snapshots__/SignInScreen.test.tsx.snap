// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 2,
      }
    }
  >
    <Text>
      Log In
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="test"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "flex": 4,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="Password123!"
      />
      <RCTSwitch
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
